wodles/gcloud/buckets/bucket.py

```diff
@@ -227,22 +244,28 @@ def get_last_creation_time():
 
                 current_creation_time = blob.time_created
 
-                if current_creation_time >= self.only_logs_after:
-                    if (current_creation_time > last_creation_time) or \
-                            (current_creation_time == last_creation_time and blob.name not in previous_processed_files):
-                        self.logger.info(f'Processing {blob.name}')
-                        processed_messages += self.process_blob(blob)
+                if not self.only_logs_after or current_creation_time >= self.only_logs_after:
+                    if last_creation_time:
+                        if current_creation_time > last_creation_time:
+                            block_process_blob()
 
-                        if current_creation_time > new_creation_time:
-                            processed_files.clear()
-                            new_creation_time = current_creation_time
-
-                        processed_files.append(blob)
+                        if current_creation_time == last_creation_time:
+                            if blob.name not in previous_processed_files:
+                                block_process_blob()
+                            else:
+                                self.logger.info(f'Skipping previously processed file: {blob.name}')
+                    elif not self.only_logs_after:
+                        if current_creation_time >= self.default_date:
+                            block_process_blob()
+                        else:
+                            self.logger.info(f'The creation time of {blob.name} is older than {self.default_date}. '
+                                             f'Skipping it...')
                     else:
-                        self.logger.info(f'Skipping previously processed file: {blob.name}')
+                        block_process_blob()
```

Comment:
If you apply the suggested change for `_get_last_creation_time` you could simplify this as you don't need to handle `None` values for last_creation_time.